lang_GENERIC_base/AST_generic.ml

```diff
@@ -140,22 +140,124 @@
  * See also pfff/lang_GENERIC/
 *)
 
+(* Provide hash_* and hash_fold_* for the core ocaml types *)
+open Ppx_hash_lib.Std.Hash.Builtin
+
 (*****************************************************************************)
 (* Accessories *)
 (*****************************************************************************)
 
 (* A set of metavariables. Access cost is O(log n). *)
 module String_set = struct
-  include Set.Make (String)
+  type t = string Set_.t
 
   type string_list = string list
   [@@deriving show]
 
   let pp fmt x =
-    pp_string_list fmt (elements x)
+    pp_string_list fmt (Set_.elements x)
+end
+
+(*****************************************************************************)
+(* Comparison and hashing *)
+(*****************************************************************************)
+
+(*
+   Create a node ID that is guaranteed to be unique within an AST,
+   as long as the whole AST is created within the same process.
+
+   Such ID should not be expected to be unique across ASTs,
+   since ASTs can be loaded from an external cache, for example.
+*)
+module Node_ID :
+sig
+  type t = private int [@@deriving show, eq, hash]
```

Comment:
This is not supported by the ocaml parser in pfff, and so semgrep choke in CI. Can you do without
this advanced feature?